### PR TITLE
Add option to link taxon to a parent in a fixture

### PR DESCRIPTION
**Pull request by Jeroen-G**
| Q               | A
| --------------- | -----
| Branch?         | 1.6
| Bug fix?        | yes
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

The taxon fixture did not allow for a parent to be set. I added the option to pass the code of an existing parent taxon to a new one with the `parent:` attribute in a fixture yaml file.

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/TaxonExampleFactory.php [hunk 1]
```diff
@@ -135,6 +140,9 @@ protected function configureOptions(OptionsResolver $resolver): void
             ->setAllowedTypes('translations', ['array'])
             ->setDefault('children', [])
             ->setAllowedTypes('children', ['array'])
+            ->setDefault('parent', function (Options $options): string {
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/TaxonExampleFactory.php]: It seems wrong. We don't want to have the same taxon as it's parent by default. The default value should be null and we should add normalizer, which will translate some `code` to taxon itself. 
```suggestion
            ->setDefault('parent', null)
            ->setAllowedTypes('parent', ['null', 'string', TaxonInterface::class])
            ->setNormalizer('parent', LazyOption::findOneBy($this->taxonRepository, 'code'))
```

**Comment by Jeroen-G** [comment]
As an example I have used the syntax in the t-shirts example fixture.

File: src/Sylius/Bundle/CoreBundle/spec/Fixture/Factory/TaxonExampleFactorySpec.php [hunk 2]
```diff
@@ -25,7 +25,7 @@
 
 class TaxonExampleFactorySpec extends ObjectBehavior
 {
-    function let(
+    public function let(
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/spec/Fixture/Factory/TaxonExampleFactorySpec.php]: All our specs don't have any `public` accessor. Can you revert this change?

Comment (Jeroen-G) [reply src/Sylius/Bundle/CoreBundle/spec/Fixture/Factory/TaxonExampleFactorySpec.php]: Done! It was the auto formatter from my editor 👀

Comment (lchrusciel) [reply src/Sylius/Bundle/CoreBundle/spec/Fixture/Factory/TaxonExampleFactorySpec.php]: but there is one more file to fix :P https://github.com/Sylius/Sylius/pull/10952/files#diff-f275e213fd9b5b34a08b735a342f79d9R28

Comment (Jeroen-G) [reply src/Sylius/Bundle/CoreBundle/spec/Fixture/Factory/TaxonExampleFactorySpec.php]: 😩 

Fixed.

**Comment by lchrusciel** [comment]
One more thing, this is a new feature, therefore we need to rebase this branch to master
